Add article email content for brief records

camillevilla opened the pull request:
Fixes part of #1448 
Related: #1552, #1553 

## Todo
- [x] add an email action for the article controller
- [x] create a partial for article brief record
- [x] rebase and utilize new EdsLinks module from Jessie's SFX panel work
- [x] get presenters / helper methods working correctly in mailers
- [x] create a partial for article full record
- [x] write tests
- [x] record links should point to `/article` not `/view`

## Requirements
### Brief record
- some cleanup to apply to current SW catalog as well as articles
  - [x]  remove plaintext 
  - [x]  remove "Message:" from the top of the email
  - [x]  create HTML links to avoid exposed URLs
- output is formatted HTML and similar to the results view:
  - [x]  structured as ordered list; sub-elements can be divs
  - [x]  title, bold, linked to SW record 
  - [x]  authors, not linked
  - [x]  composed title (journal, date, issue, etc.)
  - [x]  link to full text, with whatever wording we end up using in the search results

## Examples
### Article 
<img width="814" alt="screen shot 2017-08-04 at 12 31 54 pm" src="https://user-images.githubusercontent.com/5402927/28984081-17b800ee-7911-11e7-9a9f-9bf1c5619592.png">


### Catalog record with holdings info
<img width="436" alt="screen shot 2017-08-04 at 2 29 41 pm" src="https://user-images.githubusercontent.com/5402927/28987895-3425acca-7922-11e7-8d2f-733423b482b5.png">

camillevilla commented:
hi @jvine, can I get your review on the following brief record emails? 

### Article 
<img width="814" alt="screen shot 2017-08-04 at 12 31 54 pm" src="https://user-images.githubusercontent.com/5402927/28984081-17b800ee-7911-11e7-9a9f-9bf1c5619592.png">


### Catalog record with holdings info
<img width="430" alt="screen shot 2017-08-04 at 12 09 27 pm" src="https://user-images.githubusercontent.com/5402927/28984084-1b6f1bd2-7911-11e7-8112-f13c2ac58b9c.png">

### Catalog record with SFX link
<img width="507" alt="screen shot 2017-08-04 at 12 30 55 pm" src="https://user-images.githubusercontent.com/5402927/28984088-1f9638a8-7911-11e7-94f4-00407efc6d1e.png">
I'm currently using the link text provided by SFX. Would you prefer something like "Online"?

jvine commented:
Yay! in general. :-) 

1. The link text is fine, but maybe a label? Like 
   Online: [EBSCHOhost](link)

2. Horiz lines are not needed in the numbered brief results.

3. I wonder if the catalog holdings info would be better as nested `<ul>`s
```
<ul><li>Marine Biology Library (Miller) - Stacks
   <ul><li>SH167 .S17 G74 2010</li></ul></li>
<li>SAL3 (off-campus storage) - Stacks
   <ul><li>SH167 .S17 G74 2010</li></ul></li>
</ul>
```
Or maybe they already are? but indents would be helpful, especially for records with many items.
![screen shot 2017-08-04 at 1 34 10 pm](https://user-images.githubusercontent.com/6945691/28985937-ba7de278-7919-11e7-8a4f-3e2defd204ec.png)

camillevilla commented:
@jvine locations are currently just in `<p>` tags. Here's a shot with `<ul>` and `<li>`. What do you think?

<img width="436" alt="screen shot 2017-08-04 at 2 29 41 pm" src="https://user-images.githubusercontent.com/5402927/28987895-3425acca-7922-11e7-8d2f-733423b482b5.png">

jvine commented:
Structurally 👍 

Do you want to fuss with styling as part of this or should i make a separate ticket? I'd like to specify font to match SW and list-style/margins.

camillevilla commented:
@jvine a separate ticket, please 😥. Stay tuned for full record HTML soon...

camillevilla commented:
Hey all, I split the article full record emails off into another ticket. There's enough backend stuff going on that I'd like to get that reviewed before putting any more time into full records. 

@jvine see my remaining issue with the subject line in article brief records. Is it ok if I split that off for now? 

@jkeck When we have a chance to discuss it, my work on article full records is over in #1554.

camillevilla commented:
@jvine here's the 'Send to email' form for articles. Is this ok or would you rather I hide the `Brief record` radio button? 

![email form without full record option](https://user-images.githubusercontent.com/5402927/29044936-82a48c3a-7b76-11e7-839a-d56460f9c8af.png)

jvine commented:
@camillevilla can you suppress the radio button for now? The text will need to be updated, and I should chat with you before I write something.

camillevilla commented:
thanks @jvine! Here's our form now:
![email form without radio buttons](https://user-images.githubusercontent.com/5402927/29049304-073f6f42-7b8a-11e7-9dcb-2ca24e354a84.png)

jvine commented:
👍

camillevilla commented:
@jkeck looks like Coveralls is unhappy. I'll loop back to write some more tests this morning.

camillevilla commented:
FYI: Coveralls is happy again!

jkeck commented on app/controllers/article_controller.rb:
```diff
@@ -169,8 +170,42 @@ def eds_init
   protected
 
+  def send_emails_to_all_recipients
+    @documents = Array.wrap(@documents)
```
Are we depending on the `@documents` variable being wrapped in an array anywhere else besides the two lines below where we pass it to the mailer class?  If so, can we keep this as a local variable instead of the ivar?

camillevilla replied:
Great, that worked!

jkeck commented:
One super minor comment, other than that, this is looking good. 👍 

Excellent work!
